src/Analysis/Ast/Test/InheritNonClassTests.cs

```diff
@@ -0,0 +1,172 @@
+
+        [TestInitialize]
+        public void TestInitialize()
+            => TestEnvironmentImpl.TestInitialize($"{TestContext.FullyQualifiedTestClassName}.{TestContext.TestName}");
+
+        [TestCleanup]
+        public void Cleanup() => TestEnvironmentImpl.TestCleanup();
+
+        [TestMethod, Priority(0)]
+        public async Task InheritFromRenamedBuiltin() {
+            const string code = @"
+tmp = str
+
+class C(tmp):
+    def method(self):
+        return 'test'
+";
+            var analysis = await GetAnalysisAsync(code);
+            analysis.Diagnostics.Should().BeEmpty();
+        }
+
+
+        [TestMethod, Priority(0)]
+        public async Task InheritFromBuiltin() {
+            const string code = @"
+class C(str):
+    def method(self):
+        return 'test'
+";
+            var analysis = await GetAnalysisAsync(code);
+            analysis.Diagnostics.Should().BeEmpty();
+        }
+
+
+        [TestMethod, Priority(0)]
+        public async Task InheritFromUserClass() {
+            const string code = @"
+class D:
+    def hello(self):
+        pass
+
+class C(D):
+    def method(self):
+        return 'test'
+";
+            var analysis = await GetAnalysisAsync(code);
+            analysis.Diagnostics.Should().BeEmpty();
+        }
+
+
+        [TestMethod, Priority(0)]
+        public async Task InheritFromConstant() {
+            const string code = @"
+class C(5):
+    def method(self):
+        return 'test'
+";
+            var analysis = await GetAnalysisAsync(code);
+            analysis.Diagnostics.Should().HaveCount(1);
+
+            var diagnostic = analysis.Diagnostics.ElementAt(0);
+            diagnostic.SourceSpan.Should().Be(2, 7, 2, 8);
+            diagnostic.Message.Should().Be(Resources.InheritNonClass.FormatInvariant("5"));
+            diagnostic.ErrorCode.Should().Be(ErrorCodes.InheritNonClass);
+        }
+
+
+        [TestMethod, Priority(0)]
+        public async Task InheritFromConstantVar() {
+            const string code = @"
+x = 'str'
+
+class C(x):
+    def method(self):
+        return 'test'
+
+x = 5
+
+class D(x):
+    def method(self):
+        return 'test'
+";
+            var analysis = await GetAnalysisAsync(code);
+            analysis.Diagnostics.Should().HaveCount(2);
+
+            var diagnostic = analysis.Diagnostics.ElementAt(0);
+            diagnostic.SourceSpan.Should().Be(4, 7, 4, 8);
+            diagnostic.Message.Should().Be(Resources.InheritNonClass.FormatInvariant("str"));
+            diagnostic.ErrorCode.Should().Be(ErrorCodes.InheritNonClass);
+
+            diagnostic = analysis.Diagnostics.ElementAt(1);
+            diagnostic.SourceSpan.Should().Be(10, 7, 10, 8);
+            diagnostic.Message.Should().Be(Resources.InheritNonClass.FormatInvariant("5"));
+            diagnostic.ErrorCode.Should().Be(ErrorCodes.InheritNonClass);
+        }
+
+        [TestMethod, Priority(0)]
+        public async Task InheritFromBinaryOp() {
```

Comment:
This test is currently failing, it seems. Not showing a diagnostic is safer than showing a wrong one, though.

Reply:
Yeah, so this involved how all those tests were failing before. I tried to be too strict on what we can inherit and that caused a lot of behavior to fail. There's nothing I can really do here with binary operations, we can make the argument that a binary operation will never resolve to a class type, and under that assumption we can have a separate check for binary operations where we report them, but I'm more inclined to leave it alone and remove this test.

Reply:
That, or stick an `[Ignore]` on it to signify that the test should pass, but doesn't for now.

Reply:
Ok sounds good.